src/gluonts/_version.py

```diff
@@ -1,214 +1,244 @@
-# -*- coding: utf-8 -*-
-# This file is part of 'miniver': https://github.com/jbweston/miniver
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 #
-from collections import namedtuple
-import os
-import subprocess
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
 
-from setuptools.command.build_py import build_py as build_py_orig
-from setuptools.command.sdist import sdist as sdist_orig
+"""
+This file supports "dynamic" versions using git during development. When the
+package is built, the content of this file is replaced with a static version.
 
-Version = namedtuple("Version", ("release", "dev", "labels"))
+This is loosely based on miniver.
 
-# No public API
-__all__ = []
+There are conditions for this file to work.
 
-package_root = os.path.dirname(os.path.realpath(__file__))
-package_name = os.path.basename(package_root)
-distr_root = os.path.dirname(package_root)
-# If the package is inside a "src" directory the
-# distribution root is 1 level up.
-if os.path.split(distr_root)[1] == "src":
-    _package_root_inside_src = True
-    distr_root = os.path.dirname(distr_root)
-else:
-    _package_root_inside_src = False
+This file needs to be part of the source code. To use it, just import
+`__version__` from it::
 
-STATIC_VERSION_FILE = "_static_version.py"
-FALLBACK_RELEASE = "0.0.0"
+    # __init__.py
+    from _version import __version__
 
+The projects `setup.py` is at most `SEARCH_PACKAGE_LEVELS` directory levels up,
+since this file needs to know where it is relative to `setup.py`.
 
-def get_version(version_file=STATIC_VERSION_FILE):
-    version_info = get_static_version_info(version_file)
-    version = version_info["version"]
-    if version == "__use_git__":
-        version = get_version_from_git()
-        if not version:
-            version = get_version_from_git_archive(version_info)
-        if not version:
-            version = Version(FALLBACK_RELEASE, None, None)
-        return pep440_format(version)
-    else:
-        return version
+When using git, `setup.py` needs to be located at the root of the repository.
 
+Thus, a typical setup looks like this::
 
-def get_static_version_info(version_file=STATIC_VERSION_FILE):
-    version_info = {}
-    with open(os.path.join(package_root, version_file), "rb") as f:
-        exec(f.read(), {}, version_info)
-    return version_info
+    my_package/
```

Comment:
my_repo maybe?

Reply:
Sounds clearer, thanks. Although, the name of the two folders is often the same.